conftest.py

```diff
@@ -331,6 +378,159 @@ def imported_vm(host, vm_ref):
         logging.info("<< Destroy VM")
         vm.destroy(verify=True)
 
+@pytest.fixture(scope="function")
+def create_vms(request, host):
+    """
+    Returns list of VM objects created from `vm_definitions` marker.
+
+    `vm_definitions` marker test author to specify one or more VMs,
+    using one `dict` per VM.
+
+    Mandatory keys:
+    - `name`: name of the VM to create (str)
+    - `template`: name (or UUID) of template to use (str)
+
+    Optional keys:
+    - `vdis`: dict-specifications for VDIs (Iterable[dict[str, str]])
+      Mandatory keys:
+      - `name`
+      - `size`
+
+    Example:
+    -------
+    > @pytest.mark.vm_definitions(dict(name="vm1", template="Other install media"),
+    >                             dict(name="vm2",
+    >                                  template="CentOS 7",
+    >                                  vdis=[dict(name="vm 2 system disk",
+    >                                             size="100GiB",
+    >                                             device="xvda",
+    >                                             userdevice="0",
+    >                                             )],
+    >
+    >     ))
+    > def test_foo(create_vms):
+    >    ...
+
+    Example:
+    -------
+    > @pytest.mark.dependency(depends=["test_foo"])
+    > @pytest.mark.vm_definitions(dict(name="vm1", image_test="test_foo", image_vm="vm2"))
+    > def test_bar(create_vms):
+    >    ...
+
+    """
+    # Do we cache VMs?
+    try:
+        from data import CACHE_IMPORTED_VM
+    except ImportError:
+        CACHE_IMPORTED_VM = False
+    assert CACHE_IMPORTED_VM in [True, False]
+
+    import git
+    test_repo = git.Repo(".")
+    assert not test_repo.is_dirty(), "test repo must not be dirty to cache images"
```

Comment:
should not be tested if CACHE_IMPORTED_VM is set to False no?

Reply:
Not applicable any more: CACHE_IMPORTED_VM now has no impact on nested-host caching, which is systematic